components/api/src/main/java/com/hotels/styx/api/RequestCookie.java

```diff
@@ -51,8 +48,14 @@
      * @param value cookie value
      */
     private RequestCookie(String name, String value) {
-        checkArgument(!isNullOrEmpty(name), "name cannot be null or empty");
-        requireNonNull(value, "value cannot be null");
+        if (name == null || name.isEmpty()) {
+            throw new IllegalArgumentException("name cannot be null or empty");
+        }
+
+        if (value == null) {
```

Comment:
Java 8 has an `Objects.requireNonNull()` method to do this.
https://docs.oracle.com/javase/8/docs/api/java/util/Objects.html#requireNonNull-T-